sphinx_gallery/gen_rst.py

```diff
@@ -570,11 +578,36 @@ def generate_dir_rst(
         f"generating gallery for {build_target_dir}... ",
         length=len(sorted_listdir),
     )
-    for fname in iterator:
-        intro, title, (t, mem) = generate_file_rst(
-            fname, target_dir, src_dir, gallery_conf, seen_backrefs
+
+    parallel = list
+    p_fun = generate_file_rst
+    if gallery_conf["parallel"]:
+        from joblib import Parallel, delayed
+
+        p_fun = delayed(generate_file_rst)
+        parallel = Parallel(
+            n_jobs=gallery_conf["parallel"],
+            pre_dispatch="n_jobs",
+            batch_size=1,
+            backend="loky",
         )
+
+    results = parallel(
+        p_fun(fname, target_dir, src_dir, gallery_conf) for fname in iterator
+    )
+    for fi, (intro, title, (t, mem), out_vars) in enumerate(results):
+        fname = sorted_listdir[fi]
         src_file = os.path.normpath(os.path.join(src_dir, fname))
+        gallery_conf["titles"][src_file] = title
+        if "formatted_exception" in out_vars:
+            assert "passing" not in out_vars
+            assert "stale" not in out_vars
+            gallery_conf["failing_examples"][src_file] = out_vars["formatted_exception"]
+        elif "passing" in out_vars:
+            assert "stale" not in out_vars
+            gallery_conf["passing_examples"].append(src_file)
+        elif "stale" in out_vars:  # non-executable files have none of these three
```

Comment:
what are "these three"?

Would it be clearer to say that 'stale' examples are 'not re-executed'? or something similar?

Reply:
Changing to at the top of the three conditionals:
```
        # n.b. non-executable files have none of these three variables defined,
        # so the last conditional must be "elif" not just "else"
```